### PR TITLE
Allow whitespace before doctype

### PR DESCRIPTION
**Pull request by k4b7**
Fixes https://github.com/mozilla/slowparse/issues/84.

### CODE REVIEW

File: test/test-slowparse.js [hunk 1]
```diff
@@ -42,6 +42,14 @@ module.exports = function(Slowparse, window, document, validators) {
     });
   });
 
+  test("parsing of valid DOCTYPE with whitespace", function() {
```
Comment (Pomax) [inline test/test-slowparse.js:45]: can we also add the test cases for "comment", "white space, comment" and "comment, white space"? just to cover the spec?

**Comment by Pomax** [comment]
awesome! if we can also add the test case combinations for comments, this should be good to land.

**Comment by k4b7** [comment]
I forgot about the comment part.  Will do.

**Comment by flukeout** [comment]
@Pomax - is this good to go?

**Comment by Pomax** [comment]
@flukeout if you want to add the test cases, then yes. It really shouldn't land without those, since we have test cases for everything else, and landing code first "then we can add the tests later" typically means those test will never get added ;)

**Pull request closed without merging**